requestor-tutorials/service-development/service-example-6-external-api-request.md

```diff
@@ -0,0 +1,183 @@
+    }
+  ],
+  "compManifest": {
+    "version": "0.1.0",
+    "script": {
+      "commands": [
+        "run .*curl.*",
+      ],
+      "match": "regex"
+    },
+    "net": {
+      "inet": {
+        "out": {
+          "protocols": ["https"],
+          "urls": ["https://api.coingecko.com"]
+        }
+      }
+    }
+  }
+}
+```
+
+Created file should be [verified using JSON schema](../vm-runtime/computation-payload-manifest.md#schema-verification).
+
+Then it needs to be encoded in `base64`:
+
+```sh
+ base64 --wrap=0 manifest.json.base64 > manifest.json.base64
+```
+### 2. Yapapi example app
+
+Base64 encoded _manifest_ can be configured using [`yapapi.payload.vm.manifest`](https://yapapi.readthedocs.io/en/latest/api.html#module-yapapi.payload.manifest) function resulting in following `externa_api_request.py` file:
+
+```py
+import asyncio
+
+from yapapi import Golem
+from yapapi.services import Service
+from yapapi.payload import vm
+
+class OutboundNetworkService(Service):
+    @staticmethod
+    async def get_payload():
+        return await vm.manifest(
+            manifest = open("manifest.json.base64", "rb").read()
+            # later we may add here manifest signature, digest algorithm, and app author's certificate
+            min_mem_gib = 0.5,
+            min_cpu_threads = 0.5,
+            # capabilities used to reach Provider with a correct VM Runtime
+            capabilities=["inet", "manifest-support"],
+        )
+
+    async def run(self):
+        script = self._ctx.new_script()
+        future_result = script.run(
+            "/bin/sh",
+            "-c",
+            f"GOLEM_PRICE=`curl -X 'GET' \
+                    'https://api.coingecko.com/api/v3/simple/price?ids=golem&vs_currencies=usd' \
+                    -H 'accept: application/json' | jq .golem.usd`; \
+                echo \"Golem price: $GOLEM_PRICE USD\";",
+        )
+        yield script
+
+        result = (await future_result).stdout
+        print(result.strip() if result else "")
+
+async def main():
+    async with Golem(budget=1.0, subnet_tag="testnet") as golem:
+        await golem.run_service(OutboundNetworkService, num_instances=1)
+        await asyncio.sleep(60)
+```
+
+### 3. Verification of a request with  Computation Payload Manifest
+
+_Providers_ verify the incoming request with a _Computation Payload Manifest_ by checking if it arrives with a [signature and _App author's certificate_ signed by a certificate they trust](../vm-runtime/computation-payload-manifest.md#certificates). If there is no signature they verify if URLs used by _Computation Payload Manifest_ are [whitelisted](../../provider-tutorials/provider-cli.md#domain-whitelist).
+
+There are two ways to make our *local* _Provider_ to verify the request:
+
+- #### Whitelisting of the domain used by the app
+
+  Add `api.coingecko.com` to Provider's [domain whitelist](../../provider-tutorials/provider-cli.md#domain-whitelist):
```

Comment:
isn't coingecko whitelisted by default?

Reply:
I'd generally move the operations performed on provider nodes to a separate article to make this one clearer and shorter

Reply:
Coingecko domain will not be whitelisted because yapapi example will have a signature generated using Golem's certificate.
The example from doc shows how the app can be executed without a signature when domain is whitelisted.